Source/CLI/Main.cpp

```diff
@@ -565,11 +569,18 @@ int FFmpeg_Command(const char* FileName)
     stringstream t;
     t << MapPos++;
     Command += " -attach \"" + rawcooked_reversibility_data_FileName + "\" -metadata:s:" + t.str() + " mimetype=application/octet-stream -metadata:s:" + t.str() + " \"filename=RAWcooked reversibility data\" \"";
-    Command += FileName;
-    if (Command[Command.size() - 1] == '/' || Command[Command.size() - 1] == '\\')
-        Command.pop_back();
-    Command += ".mkv\"";
-
+    if (OutputFileName.empty())
+    {
+        Command += FileName;
+        if (Command[Command.size() - 1] == '/' || Command[Command.size() - 1] == '\\')
+            Command.pop_back();
+        Command += ".mkv";
+    }
+    else
+    {
+        Command += OutputFileName;

```

Comment:
One could choose e.g. `my_file.mxf` (any container different from `.mkv`). Would this not be a problem, as the file type `.mkv` is the only discriminant used by default to differentiate between a file to **encode** with RAWcooked and a file to **decode** with RAWcooked?

Reply:
Correct. I missed that.
I added a check on the file name extension.
Note: in the future, more output formats could be supported, when RAWcooked implement the decode, the reversibility file is not tied to Matroska (or FFV1), but for the moment there is only Matroska (and FFV1) support for decoding, true.

Reply:
>I added a check on the file name extension.

+1